pandas/tests/io/test_excel.py

```diff
@@ -119,11 +134,11 @@ def get_exceldf(self, basename, ext, *args, **kwds):
 class ReadingTestsBase(SharedItems):
     # This is based on ExcelWriterBase
 
-    @pytest.fixture(autouse=True, params=['xlrd', None])
-    def set_engine(self, request):
+    @pytest.fixture(autouse=True)
```

Comment:
I removed the use of engine as a parametrized fixture, in favour of setting it as a class attribute of derived, engine specific, classes

Reply:
This should be reverted and just parametrize on `openpyxl` as well

Reply:
Still want to revert this. We prefer parametrization to creating subclasses for testing, and this also had the side effect of reducing coverage on None

Reply:
Ok, will give it a go

Reply:
Still not clear on what the point of removing this was - we moved away from subclasses previously so this takes us backwards. This also removes coverage for `engine=None` - what issue was this causing that required reverting to subclasses?